### ISSUE TITLE
Simplify configuration to run Authelia in demo mode

### ISSUE THREAD

**Issue by clems4ever** [clems4ever/authelia#233]
The objective is to improve adhesion of Authelia by giving a way to users to test it without setting the whole environment. For this configuration must be reduced to the strict minimum (without ldap, redis, mongo). 

It will also improve testability since some integration tests will be able to run without the entire environment.

Obviously, this won't be meant for production environments.

**Comment by tsukasagenesis** [comment]
It's definitely a great idea, I start to install it but when I saw LDAP it's instantly a turn off. Between redis, mongo, and ldap I had to install more than 200MB for a small project of secure my dev env. 

Please make it append,  I will be glad to try it out.

**Comment by starouille** [comment]
same here, i will use Authelia with less dependencies.

**Comment by clems4ever** [comment]
Good to know. For your info, only the LDAP server is mandatory, everything else (persistent storage and server session storage) can be stored either in memory or on disk.

For persistent storage, the option is in the config file and for session storage I think that removing the *redis* object from the configuration file makes Authelia use the Memory backend of express-session (this was previously used for testing purpose and I think I did not remove the code).

**Comment by starouille** [comment]
yes, i already saw "local storage".  LDAP is the major blocking point because we didn't have it.

**Comment by clems4ever** [comment]
Clear, I will add support for a password database then.
Thanks.

**Comment by clems4ever** [comment]
@starouille, I think I will use htpasswd files for the password DB. What do you think?

**Comment by starouille** [comment]
htpasswd with bcrypt hashed password is perfect :).

**Comment by clems4ever** [comment]
Hello, #262 does the job.
